integration_test/ops_agent_test.go

```diff
@@ -465,74 +465,95 @@ func TestHTTPRequestLog(t *testing.T) {
 			"status":        200,
 		}
 
-		// Write a log with an http request in the message to cloud logging and immediately query it.
-		writeAndQueryLog := func(httpRequestKey string, logId string) *cloudlogging.Entry {
-			logBody := map[string]interface{}{
-				"logId":        logId,
-				httpRequestKey: httpRequestBody,
-			}
-			logBytes, err := json.Marshal(logBody)
-			if err != nil {
-				t.Fatalf("could not marshal test log: %v", err)
-			}
-			err = gce.UploadContent(
-				ctx,
-				logger,
-				vm,
-				strings.NewReader(string(logBytes)+"\n"),
-				logPath)
-			if err != nil {
-				t.Fatalf("error writing log line: %v", err)
-			}
+		// Log with HTTP request data nested under "logging.googleapis.com/httpRequest".
+		// const newHTTPRequestKey = confgenerator.HttpRequestKey
+		const newHTTPRequestKey = "logging.googleapis.com/http_request"
+		const newHTTPRequestLogId = "new_request_log"
+		newLogBody := map[string]interface{}{
+			"logId":           newHTTPRequestLogId,
+			newHTTPRequestKey: httpRequestBody,
+		}
+		newLogBytes, err := json.Marshal(newLogBody)
+		if err != nil {
+			t.Fatalf("could not marshal new test log: %v", err)
+		}
+
+		// Log with HTTP request data nested under "logging.googleapis.com/http_request".
+		// const oldHTTPRequestKey = "logging.googleapis.com/http_request"
+		const oldHTTPRequestKey = confgenerator.HttpRequestKey
+		const oldHTTPRequestLogId = "old_request_log"
+		oldLogBody := map[string]interface{}{
+			"logId":           oldHTTPRequestLogId,
+			oldHTTPRequestKey: httpRequestBody,
+		}
+		oldLogBytes, err := json.Marshal(oldLogBody)
+		if err != nil {
+			t.Fatalf("could not marshal old test log: %v", err)
+		}
+
+		// Write both logs to log source file at the same time.
+		err = gce.UploadContent(
+			ctx,
+			logger,
+			vm,
+			strings.NewReader(fmt.Sprintf("%s\n%s\n", string(newLogBytes), string(oldLogBytes))),
+			logPath)
+		if err != nil {
+			t.Fatalf("error writing log line: %v", err)
+		}
 
-			entry, err := gce.QueryLog(
+		queryLogById := func(logId string) (*cloudlogging.Entry, error) {
+			return gce.QueryLog(
 				ctx,
 				logger.ToMainLog(),
 				vm,
 				"mylog_source",
 				time.Hour,
 				fmt.Sprintf("jsonPayload.logId=%q", logId),
 				gce.QueryMaxAttempts)
-			if err != nil {
-				t.Fatalf("could not find written log with id %s: %v", logId, err)
-			}
-			return entry
 		}
 
-		// Test that the new documented field, "logging.googleapis.com/httpRequest", will be
-		// parsed as expected by Fluent Bit.
-		t.Run("parse new HTTPRequest key", func(t *testing.T) {
-			httpRequestKey := confgenerator.HttpRequestKey
-			entry := writeAndQueryLog(httpRequestKey, "341231")
+		isKeyInPayload := func(httpRequestKey string, entry *cloudlogging.Entry) bool {
 			payload := entry.Payload.(*structpb.Struct)
+			foundKey := false
 			for key := range payload.GetFields() {
 				if key == httpRequestKey {
-					t.Fatal("expected request key to be stripped out of message")
+					foundKey = true
```

Comment:
Since you made it a separate function, this can just be `return true`, i.e.:
```go
		isKeyInPayload := func(httpRequestKey string, entry *cloudlogging.Entry) bool {
			payload := entry.Payload.(*structpb.Struct)
			for key := range payload.GetFields() {
				if key == httpRequestKey {
					return true
				}
			}
			return false
		}
```

[Optional] Go style encourages single-letter loop variable names, so you could even use `for k := range payload.GetFields()`.

Reply:
Implemented in [59b0a82](https://github.com/GoogleCloudPlatform/ops-agent/pull/655/commits/59b0a8260d13e87b6f495b888cd28d5f2e8866c3)